service/handlers.go

```diff
@@ -39,12 +42,18 @@ func handleQuery(c *Core, w *ResponseWriter, r *Request) {
 		w.Error(zqe.ErrInvalid(err))
 		return
 	}
-	d, err := queryio.NewDriver(zio.NopCloser(w), format, !noctrl)
+	d, err := queryio.NewDriver(w.ResponseWriter, format, !noctrl)
 	if err != nil {
 		w.Error(err)
 	}
+	var statsTicker <-chan time.Time
+	if !noctrl {
+		t := time.NewTicker(queryStatsInterval)
+		statsTicker = t.C
+		defer t.Stop()
+	}
 	defer d.Close()
```

Comment:
Nit: Move defers closer to their corresponding assignments to keep idioms intact.
```suggestion
	d, err := queryio.NewDriver(w.ResponseWriter, format, !noctrl)
	if err != nil {
		w.Error(err)
	}
	defer d.Close()
	var statsTicker <-chan time.Time
	if !noctrl {
		t := time.NewTicker(queryStatsInterval)
		defer t.Stop()
		statsTicker = t.C
	}
```